daml-assistant/daml-helper/src/DA/Daml/Helper/Ledger.hs

```diff
@@ -0,0 +1,51 @@
+-- Copyright (c) 2019 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+-- SPDX-License-Identifier: Apache-2.0
+module DA.Daml.Helper.Ledger (
+    HostAndPort(..),
+    listParties, PartyDetails(..), Party(..),
+    lookupParty,
+    allocateParty,
+    uploadDarFile,
+    ) where
+
+import DA.Ledger(LedgerService,PartyDetails(..),Party(..))
+import Data.List.Extra as List
+import Data.String(fromString)
+import qualified DA.Ledger as L
+import qualified Data.ByteString as BS
+import qualified Data.Text.Lazy as Text(pack)
+
+data HostAndPort = HostAndPort { host :: String, port :: Int }
+
+instance Show HostAndPort where
+    show HostAndPort{host,port} = host <> ":" <> show port
+
+listParties :: HostAndPort -> IO [PartyDetails]
+listParties hp = run hp L.listKnownParties
+
+lookupParty :: HostAndPort -> String -> IO (Maybe Party)
+lookupParty hp name = do
+    xs <- listParties hp
+    let text = Text.pack name
+    let pred PartyDetails{displayName,party} = if text == displayName then Just party else Nothing
+    return $ List.firstJust pred xs
+
+allocateParty :: HostAndPort -> String -> IO Party
+allocateParty hp name = run hp $ do
+    let text = Text.pack name
+    let request = L.AllocatePartyRequest
+            { partyIdHint = "" -- text -- ???
```

Comment:
Yes we should set the party name as a hint. Sandbox can take it as is and in Canton we would get more readable party ids.

Reply:
Why do we even care that party names are readable?

Reply:
It makes debugging easier if the party ids are readable to some degree. I'd appreciate if we set the hint based on the name.

Reply:
done.. next PR